### PR TITLE
SPARK-5393. Flood of util.RackResolver log messages after SPARK-1714

### PR DESCRIPTION
**Pull request by sryza**
Previously I had tried to solve this with by adding a line in Spark's log4j-defaults.properties.

The issue with the message in log4j-defaults.properties was that the log4j.properties packaged inside Hadoop was getting picked up instead. While it would be ideal to fix that as well, we still want to quiet this in situations where a user supplies their own custom log4j properties.

### CODE REVIEW

File: yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala [hunk 1]
```diff
@@ -60,6 +62,9 @@ private[yarn] class YarnAllocator(
 
   import YarnAllocator._
 
+  // RackResolver logs an INFO message whenever it resolves a rack, which is way too often.
+  Logger.getLogger(classOf[RackResolver]).setLevel(Level.WARN)
```
Comment (vanzin) [inline yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala]: Can you explain in the commit message why the log4j configuration doesn't work? From just peeking at the code it seems like both approaches should be the same.

Comment (sryza) [inline yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala]: The explanation is a little involved, so will explain here:
The issue with the message in log4j-defaults.properties was that the log4j.properties packaged inside Hadoop was getting picked up instead.  While it would be ideal to fix that as well, we still want to quiet this in situations where a user supplies their own custom log4j properties.

Comment (vanzin) [inline yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala]: Can you mention that in your first comment so its picked up as the commit message? (That's what I meant by "commit message", btw.)

Also, I'm pretty sure I've seen a bug to fix the config file issue. Might be worth it to look for it and mention it in the comments.

Comment (sryza) [inline yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala]: Ah makes sense, placed it there.  Looked a bit for that bug, but couldn't find it.

File: core/src/main/scala/org/apache/spark/SparkContext.scala [hunk 2]
```diff
@@ -1982,7 +1982,7 @@ object SparkContext extends Logging {
       case "yarn-client" =>
         val scheduler = try {
           val clazz =
-            Class.forName("org.apache.spark.scheduler.cluster.YarnClientClusterScheduler")
+            Class.forName("org.apache.spark.scheduler.cluster.YarnScheduler")
```
Comment (tgravescs) [inline core/src/main/scala/org/apache/spark/SparkContext.scala]: why this change?

File: yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala (hunk 1 above)
Comment (ksakellis) [inline yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala]: So won't this override any user supplied loglevel for RackResolver? What if the user legitimately wants to see the messages?

Comment (sryza) [inline yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala]: It will.  Which isn't ideal, but I don't think it's a big deal.

Comment (ksakellis) [inline yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala]: Well, I disagree. A user will get very frustrated if they are debugging an issue and they can't turn on the logging. Can you add a check: Logger.getLogger(classOf[RackResolver]).getLevel() != null at least you won't be overriding the logging level if it is set.

**Comment by tgravescs** [comment]
I ran into this on spark-shell. Definitely annoying.  I tried the patch and and it works. I was also able to enable the RackResolver logging back to INFO.

+1. Thanks @sryza
